### PR TITLE
rust: read: handle padded compressed chunks properly

### PR DESCRIPTION
**Pull request by james-rms**
### Changelog
- rust: fixed case where a compressed chunk with padding bytes added by the compressor at the end would break decompression.

### Docs

None.

### Description

When MCAPs are written, the compressor may insert padding bytes at the end of a compressed block after all useful data bytes are written. This means that when the decompressor is reading messages out, it will have decompressed all message bytes out of a compressed chunk _before_ it reaches the end of the compressed data. Right now, the rust MCAP library will see that there is more unused data in the compressed chunk, try to decompress another message, but get no output from of the decompressor when it tries.  Right now this causes the reader to loop infinitely. This PR adds a file to LFS which triggers this condition and fixes it.

<table><tr><th>Before</th><th>After</th></tr><tr><td>

</td><td>

</td></tr></table>

### CODE REVIEW

File: rust/src/sans_io/read.rs [hunk 1]
```diff
@@ -1101,4 +1102,31 @@ mod tests {
         }
         Ok(())
     }
+
+    #[test]
+    fn test_decompression_does_not_fail() {
+        let mut f = std::fs::File::open("tests/data/break_zstd_decompression.mcap")
```
Comment (gasmith) [inline rust/src/sans_io/read.rs:1108]: Nit: Could we rename this test and/or mcap file to more clearly indicate the corner case that it's tickling?

**Review by gasmith — APPROVED** [review]
LG
